<launch>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="paused" default="false"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find champ_gazebo)/worlds/outdoor.world"/>
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="headless" value="$(arg headless)"/>
    </include>

    <include file="$(find champ_description)/launch/description.launch"/>
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" args="-urdf -model champ -param robot_description -z 0.15" />
    
    <rosparam file="$(find champ_gazebo)/config/champ.yaml" command="load"/>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="
        --namespace=champ
        joint_states_controller
        joint_group_position_controller
        ">
    </node>

    <node pkg="champ_gazebo" name="champ_joints_controller" type="joints_controller.py" output="screen"/>
    <node name="odometry_ground_truth_tf" pkg="champ_gazebo" type="odometry.py" output="screen" />
    <node name="contact_sensor" pkg="champ_gazebo" type="contact_sensor.py" output="screen" />

    <node name="depthimage_to_laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" output="screen">
        <remap from="image" to="/camera1/depth/image_raw"/>
        <remap from="camera_info" to="/camera1/depth/camera_info"/>
        <param name="output_frame_id" value="camera_link" />
        <param name="scan_height" value="100" />
    </node>
</launch>

